plugin/src/sbt-test/sbtcompendium/basic/build.sbt

```diff
@@ -7,34 +7,34 @@ scalaVersion := "2.12.10"
 
 version := "0.1"
 
-compendiumProtocolIdentifiers := List("MyProtocol")
-compendiumServerHost := "localhost"
-compendiumServerPort := 8080
+compendiumSrcGenProtocolIdentifiers := List("MyProtocol")
+compendiumSrcGenServerHost := "localhost"
+compendiumSrcGenServerPort := 8080
 
 sourceGenerators in Compile += Def.task {
-  compendiumGenClients.value
+  compendiumSrcGenClients.value
 }.taskValue
 
-def generateClient(target: IdlName, identifier: String): IO[String] =
+def generateClient(target: IdlName, identifier: String): IO[List[String]] =
   IO(
-    """
+    List("""
     package higherkindness.compendium.storage
     object TestFile extends App {
       println("Hey")
     }
-    """.stripMargin
+    """.stripMargin)
   )
 
-compendiumGenClients := {
+compendiumSrcGenClients := {
 
-  val generateProtocols = compendiumProtocolIdentifiers.value.toList.map { protocolId =>
-    val targetFile       = (sbt.Keys.sourceManaged in Compile).value / "compendium" / s"$protocolId.scala"
-    val generateProtocol = CompendiumUtils.generateCodeFor(protocolId, targetFile, generateClient)
+  val generateProtocols = compendiumSrcGenProtocolIdentifiers.value.toList.map { protocolId =>
+    def targetFile(id: String) = (sbt.Keys.sourceManaged in Compile).value / "compendium" / s"$protocolId.scala"
```

Comment:
The `id` is ignored, isn't it? Can we revert it back to a `val`?

Reply:
In the example is not needed but expected by the `generateCodeFor` function, but  in nested avro case, it generates one file per class so we need to add an id to not overwrite it